gpt_index/logger/base.py

```diff
@@ -0,0 +1,41 @@
+"""Logger class."""
+
+from typing import Dict
+
+from pydantic import BaseModel
+
+
+class LlamaLogger:
+    """Logger class."""
+
+    def __init__(self):
+        """Init params."""
+        self._logs = []
+        self._metadata = {}
+
+    def reset(self) -> None:
+        """Reset logs."""
+        self._logs = []
+
+    def set_metadata(self, metadata: Dict) -> None:
+        """Set metadata."""
+        self._metadata.update(metadata)
+
+    def unset_metadata(self, metadata_keys: Dict) -> None:
```

Comment:
Is there a reason for metadata_keys to be a `dict` here rather than a `set` or `list`?

Reply:
oh they should be a set :)

Reply:
good catch!